RELEASES.md

```diff
@@ -1,3 +1,97 @@
+Version 1.70.0 (2023-06-01)
+==========================
+
+<a id="1.70.0-Language"></a>
+
+Language
+--------
+- [Relax ordering rules for `asm!` operands](https://github.com/rust-lang/rust/pull/105798/)
+- [Properly allow macro expanded `format_args` invocations to uses captures](https://github.com/rust-lang/rust/pull/106505/)
+- [Lint ambiguous glob re-exports](https://github.com/rust-lang/rust/pull/107880/)
+- [Initial support for return type notation (RTN)](https://github.com/rust-lang/rust/pull/109010/)
+
+<a id="1.70.0-Compiler"></a>
+
+Compiler
+--------
+- [Extend -Cdebuginfo with new options and named aliases](https://github.com/rust-lang/rust/pull/109808/)
+  This provides a smaller version of debuginfo for cases that only need line number information
+  (`-Cdebuginfo=line-tables-only`), which may eventually become the default for `-Cdebuginfo=1`.
+- [Introduce a no-op `PlaceMention` statement for `let _ =`.](https://github.com/rust-lang/rust/pull/102256/)
+  FIXME: Unclear what, if any, user-visible impact this had on landing. Clarify with T-compiler.
```

Comment:
@cjgillot, @lcnr can you clarify here? PR has a bunch of discussion and some lang design meeting references but not clear on what the final state was.

Reply:
```rust
pub fn totally_safe(ptr: *const u8) {
    let _ = *ptr;
}
```
This code used to compile on stable, but does not compile anymore: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=10e38c3a7691cefc7830828affc6f8cf

Reply:
So this should be put under compatibility notes as well.

Reply:
tried to explain the changes in https://github.com/rust-lang/rust/pull/102256#issuecomment-1462114626

> Is a (mostly theoretical) breaking change for the following 2 snippets:
> 
> const checking:
> 
> ```rust
> const unsafe fn bad_const_unsafe_deref_raw_underscore(x: *mut usize) {
>     let _ = *x; // Now also errors with this PR
> }
> 
> const unsafe fn already_errored(x: *mut usize) {
>     let _a = *x; // ERROR dereferencing raw mutable pointers in constant functions is unstable
> }
> ```
> 
> unsafety checking:
> 
> ```rust
> fn bad_deref_raw_underscore(x: *mut usize) {
>     let _ = *x; // Now also errors with this PR
> }
> 
> fn already_errored(x: *mut usize) {
>     let _a = *x; // ERROR dereference of raw pointer is unsafe
> }
> ```